ignite/cmd/network_chain_show_genesis.go

```diff
@@ -55,59 +55,30 @@ func networkChainShowGenesisHandler(cmd *cobra.Command, args []string) error {
 		return err
 	}
 
-	genesisPath, err := c.GenesisPath()
+	// generate the genesis in a temp dir
+	tmpHome, err := os.MkdirTemp("", "*-spn")
 	if err != nil {
 		return err
 	}
-
-	spnChainID, err := n.ChainID(cmd.Context())
-	if err != nil {
+	defer os.RemoveAll(tmpHome)
+
+	c.SetHome(tmpHome)
+
+	if err := prepareFromGenesisInformation(
+		cmd,
+		cacheStorage,
+		launchID,
+		n,
+		c,
+		chainLaunch,
+	); err != nil {
 		return err
 	}
 
-	// check if the genesis already exists
-	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
-		// fetch the information to construct genesis
-		genesisInformation, err := n.GenesisInformation(cmd.Context(), launchID)
-		if err != nil {
-			return err
-		}
-
-		// create the chain in a temp dir
-		tmpHome, err := os.MkdirTemp("", "*-spn")
-		if err != nil {
-			return err
-		}
-		defer os.RemoveAll(tmpHome)
-
-		c.SetHome(tmpHome)
-
-		rewardsInfo, lastBlockHeight, unboundingTime, err := n.RewardsInfo(
-			cmd.Context(),
-			launchID,
-			chainLaunch.ConsumerRevisionHeight,
-		)
-		if err != nil {
-			return err
-		}
-
-		if err = c.Prepare(
-			cmd.Context(),
-			cacheStorage,
-			genesisInformation,
-			rewardsInfo,
-			spnChainID,
-			lastBlockHeight,
-			unboundingTime,
-		); err != nil {
-			return err
-		}
-
-		// get the new genesis path
-		genesisPath, err = c.GenesisPath()
-		if err != nil {
-			return err
-		}
+	// get the new genesis path
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return err
 	}
 
 	if err := os.MkdirAll(filepath.Dir(out), 0744); err != nil {
```

Comment:
I think we can track it in another issue since it would imply replacing all usage of `os.Rename()` (at least where `tmp` is used) and therefore becomes out of scope of this PR

Reply:
https://github.com/ignite/cli/issues/2660

Reply:
@tbruyelle which command is failing for you because of this?

Maybe we can also work directly on this issue if this prevent you from testing the feature